backends/core/workers.py

```diff
@@ -1842,3 +1843,65 @@ def _set_target_column(self, client, dataset_resource_name, target_column_spec):
     response = client.projects().locations().datasets() \
                      .patch(name=dataset_resource_name, body=body).execute()
     self.log_info('Modified target column for dataset: %s', response)
+
+class AdsDataHubWorker(Worker):
+  """Abstract AdsDataHub worker."""
+
+  GLOBAL_SETTINGS = ['developer_token']
+  ADH_SCOPE = 'https://www.googleapis.com/auth/adsdatahub'
+
+  def _get_adh_client(self):
+    credentials = ServiceAccountCredentials.from_json_keyfile_name(_KEY_FILE,
+                                                                   scopes=[ADH_SCOPE])
+    self._adh_client = build('adsdatahub',
+                             'v1',
+                             credentials=credentials,
+                             developerKey=self._params['developer_token'].strip())
+
+class AdsDataHubQueryLauncher(AdsDataHubWorker):
+  """AdsDataHub worker that launches a pre-existing query."""
+
+  PARAMS = [
+    ('customer_id', 'string', True, '', 'Customer Id'),
+    ('query_id', 'string', True, '', 'Query Id'),
+    ('payload', 'text', True, '', 'Payload')
+  ]
+
+  def _execute(self):
+    self._get_adh_client()
+
+    operation = self._adh_client.customers().analysisQueries().start(
+        name='customers/{customer_id}/analysisQueries/{query_id}'
+          .format(
+              query_id=self._params['query_id'],
+              customer_id=self._params['customer_id']),
+          body=json.loads(self._params['payload'])
```

Comment:
This should be aligned with `name=` I believe

Reply:
I aligned `body=` with `name=` but kept the other indented to the right - hope it makes sense for you too